src/watch_filter.rs

```diff
@@ -173,15 +200,54 @@ impl Filterer for WatchFilter {
         }
 
         // By process of elimination the event is valid
-        tracing::trace!("passing event: {event:?}");
+        tracing::debug!("passing event: {event:?}");
         Ok(true)
     }
 }
 
+#[derive(Debug)]
+pub struct Config {
+    pub manifest_pattern: WatchPattern,
+    pub source_patterns: Vec<WatchPattern>,
+    pub artifact_patterns: Vec<WatchPattern>,
+    pub ignore_patterns: Vec<WatchPattern>,
+}
+
+pub struct WatchPattern {
+    pub active: bool,
```

Comment:
When reading both the consuming code and the generation code, I found it a bit hard to follow what `active` was doing.  I got the impression that `active = false` meant "don't watch this," and that it was a permanent state, i.e. not temporarily moved inactive then reactivated again.  But if that's the case, why have the pattern at all?  Why not just leave it out?  So I'm clearly missing something (perhaps relating to how `active` is sometimes checked and sometimes not).  It would be good to comment the `active` field and/or rename it to make the intended use clearer.

Reply:
Good call out. `active` is in fact permanent. A better name is `should_watch`. The reason we need this field is because we want to have all possible patterns in the filter so that we can match them in the `matches_*_pattern` methods. However, we don't always want to pay attention to changes in these patterns hence `should_watch`.

Reply:
Renamed and commented.